File: backend/app/schemas/destination.py
from typing import Optional, List
from pydantic import BaseModel


class TableColumn(BaseModel):
    """Schema for table column metadata."""
    name: str
    type: str
    nullable: bool
    default: Optional[str] = None


class TableSchema(BaseModel):
    """Schema for database table structure."""
    schema_name: str
    table_name: str
    columns: List[TableColumn]

File: frontend/src/types/destination.ts
import type { LoadStrategy } from './etl-job'

export interface TableColumn {
  name: string
  type: string
  nullable: boolean
  default?: string
}

export interface TableSchema {
  schema_name: string
  table_name: string
  columns: TableColumn[]
}

export interface DestinationConfig {
  credentialId: number
  schema: string
  tableName: string
  loadStrategy: LoadStrategy
  upsertKeys?: string[]
  createNewTable?: boolean
  newTableDDL?: string
}
